pkgs/tools/inputmethods/ibus-engines/ibus-mozc/default.nix

```diff
@@ -1,89 +1,73 @@
-{ lib, stdenv, fetchFromGitHub, which, ninja, pkg-config, protobuf
-, ibus, gtk2, zinnia, qt5, libxcb, tegaki-zinnia-japanese, python3Packages }:
-
-stdenv.mkDerivation rec {
+{ lib
+, buildBazelPackage
+, fetchFromGitHub
+, qt6
+, pkg-config
+, bazel
+, ibus
+, unzip
+, xdg-utils
+}:
+
+buildBazelPackage rec {
   pname = "ibus-mozc";
-  version = "2.26.4660.102";
+  version = "2.29.5268.102";
 
   src = fetchFromGitHub {
     owner = "google";
     repo = "mozc";
-    rev = "refs/tags/${version}";
-    hash = "sha256-sgsfJZALpPHFB5bXu4OkRssViRDaPcgLfEyGhbqvJbI=";
+    rev = version;
+    hash = "sha256-B7hG8OUaQ1jmmcOPApJlPVcB8h1Rw06W5LAzlTzI9rU=";
     fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [
-    which ninja python3Packages.python python3Packages.six
-    python3Packages.gyp pkg-config qt5.wrapQtAppsHook
-  ];
+  nativeBuildInputs = [ qt6.wrapQtAppsHook pkg-config unzip ];
 
-  buildInputs = [ protobuf ibus gtk2 zinnia qt5.qtbase libxcb ];
+  buildInputs = [ ibus qt6.qtbase ];
 
-  postUnpack = lib.optionalString stdenv.isLinux ''
-    sed -i 's/-lc++/-lstdc++/g' $sourceRoot/src/gyp/common.gypi
-  '';
+  dontAddBazelOpts = true;
+  removeRulesCC = false;
 
-  configurePhase = ''
-    runHook preConfigure
+  inherit bazel;
 
-    export GYP_DEFINES="document_dir=$out/share/doc/mozc use_libzinnia=1 use_libprotobuf=1 ibus_mozc_path=$out/lib/ibus-mozc/ibus-engine-mozc zinnia_model_file=${tegaki-zinnia-japanese}/share/tegaki/models/zinnia/handwriting-ja.model"
-    cd src && python build_mozc.py gyp --gypdir=${python3Packages.gyp}/bin --server_dir=$out/lib/mozc
+  fetchAttrs = {
+    sha256 = "sha256-17QHh1MJUu8OK/T+WSpLXEx83DmRORLN7yLzILqP7vw=";
```

Comment:
I recommend to add     `disallowedReferences = [ "/nix/store" ]; `

this check fails if you remove preInstall

Reply:
on a second though, maybe its better to add it to buildBazelPackage depending how much more work we want to do. I'll leave it upto you to decide

Reply:
I think empty [`allowedReferences`](https://nixos.org/manual/nix/stable/language/advanced-attributes.html#adv-attr-allowedReferences) is more idiomatic. I have not seen disallowing whole `/nix/store` documented anywhere.

I do not think it is feasible to do it in the outer derivation, since there are ELF files that link other libraries.

Reply:
we are trying to do it on FOD, to prevent references to build inputs. what would empty allowedReferences do ?

Reply:
appologies, i just read documentation :-) cc @pineapplehunter

Reply:
Because the result is in a tar.gz it seems like it's not referencing anything.

```shell
# this only shows one reference
$ nix path-info ".#ibus-engines.mozc.deps" -r
/nix/store/p75bdk45s2ncjfqskhh8xwp47103hz49-ibus-mozc-2.29.5268.102-deps.tar.gz
```

Adding allowedReferences to fetchAttrs didn't seem to help in this case. Is there any simple way to check for references in maybe postInstal?

Reply:
```
error: hash mismatch in fixed-output derivation '/nix/store/9l16176d0m63j51x6ga10mlid605dzkq-ibus-mozc-2.29.5268.102-deps.tar.gz.drv':
         specified: sha256-17QHh1MJUu8OK/T+WSpLXEx83DmRORLN7yLzILqP7vw=
            got:    sha256-cONjTaGQcCMhfjLCPa2BSLE9/bVHvFOuv06Pg+wlJm4=
```

Reply:
Thanks!
So this time, it seems that some of the bazel deps (`zip_code_ken_all` and `zip_code_jigyosyo`) changed. It was in the logs before, but I assumed those files don't change so frequently.
I will try to do something like what's done in #251706.

Reply:
This problem should be fixed now. I replaced all the http urls with a file path from a github repo.
Let's wait a few more days and check again if anything fails.